hw/application_fpga/core/uds/README.md

```diff
@@ -31,4 +31,15 @@ Any access to another address will be ignored by the core.
 
 ## Implementation
 
-The UDS words are implemented using discrete registers.
+The 8 UDS words are stored using 32 named LUT4s identified by
+"uds\_rom\_idx".
+
+We use bit indexing from all of the 32 LUTs to store each bit of the
+word, i.e., the first UDS word consists of bit 0 from each 32 LUTs,
+the second word consists of bit 1 from each of the 32 LUTs, and so
+on... for a maximum storage of 16 words.
+
+The most significant bit in the LUT address serves as the read-enable
+input. This requires the lower half of initialization bits to be
+forced to zero, ensuring that the memory outputs zero when the
+read-enable signal is inactive.
```

Comment:
Possibly not talk about read enable. We want to be able to read UDS words once per power cycle. Since the lower 8 bits in each LUT4 contains the UDS words, the upper 8 bits are set to zero. By setting bit four in the LUT4 address we force reading of bits in the LUT4 to this area, thus getting all zero data returned to the CPU. Or something similar.

We are in fact reading and the read enable (RE) signal is a bad naming. It will be changed. But with the suggested rephrasing such a change will not affect the description.